### ISSUE TITLE
Project status questions

### ISSUE THREAD

**Issue by dpagini** [acquia/lift-sdk-php#25]
Just wondering, I see this project hasn't been updated in about 5+ months... what is the status of this?

I downloaded the code and it _seems_ to be working for me. I need to keep exploring it a little to see if it's viable for me to use....

Is there any chance this could have a tagged release, even if it's a pre-release? It's also not on packagist at the moment, so that would be a help too.

On a related note, is the acquia lift API versioned at all? It would be good to associate this SDK with a Lift API version so we could tell it's the latest. I guess I just find myself confused by all the Lift API documentation.

**Comment by ynx** [comment]
@dpagini, thanks for checking in! lift-sdk-php is the current preferred SDK to connect to Lift Decision services. As much as I know, we have fixed all known issues. Several internal customers use this library.

Personally I think we should tag a release. Depending on product vision, this could still be a v0.1 or already v1.0. Let me bring in our Product Owner see what he thinks. @tottey what do you say?

Acquia Lift's Decision Engine is not versioned at the moment. We have started off building the service without, and haven't found a pressing practical need yet; we have invested in other enhancements. 

Sorry that the documentation confuses you! If you could let us know which parts could be better written that would be great, we have documentation specialists working with the team to keep documentation easy and up to date.

**Comment by dpagini** [comment]
Thanks @ynx! +1 to the tag, obviously! =P

Ok, so I came across [this page](http://docs.decision-api.acquia.com/) which says it's API v1. Notice the `/capture` endpoint here, with just the POST option.

Then I came across [this page](http://docs.lift.acquia.com/decision/), also v1, and now capture has POST and GET options.

So right there, it's just hard to tell which of these documentation sites I'm supposed to be using, which just gives me hesitation as to the state of the API. 

So I just submitted a question into some Lift support folks around using the decision API to accomplish the same thing as the `captureIdentity()` javascript call. I want to, server side, tell Lift the email address of a user whose `tc_ptid` cookie value that I know. I am having a hard time from the documentation of the `/capture` endpoint to tell how I can accomplish this... which parameters do I need to send along, and then how does that translate to this SDK.

Thanks for your quick response!

**Comment by ynx** [comment]
Did you find http://docs.decision-api.acquia.com/ by Googling? I see it is still indexed. It is not a up-to-date URL or publication, and we should really take it offline plus remove all references. @tottey / @qw1hbmk do we have a clean up story there somewhere in the backlog?

David, please use http://docs.lift.acquia.com/decision/. Sorry for the confusion.

And yes you are right, I can see the doc is marked v1. We haven't implemented "versioning" in API, meaning we haven't started using any URL suffix, request headers, or etc to allow the API user to identify his/her request's version.

I didn't participate this library's development but I have tested in the past, so I'd still like to offer insights. If I were you, I would look at [this capture example](https://github.com/acquia/lift-sdk-php/blob/d70e8ac1a25fd2ba8f664cba86912f2e5c2bf03f/examples/Capture.php#L33). With in there, the `$capturePayload`'s settings seems to be very interesting. Reading [capturePayload's class](https://github.com/acquia/lift-sdk-php/blob/0ce5c081ff43a674d49023a82b9b1fd7e10b2db3/src/Entity/CapturePayload.php#L7), in the comments it explains how each function can be used, for example, `setIdentitySource()` and `setIdentity()` seem to be able to let you do `setIdentitySource('email')` and `setIdentity('customer.email.address@their-domain.com')`. If I were you, I would try using the same Capture.php example but with actual Decision API url, keys, account, and site id, and just see if I can `ping()` through, then set email identity. This is a standard use so I wouldn't expect any big surprises.

Good luck! ^^

**Comment by tottey** [comment]
@dpagini: Yes, please let us know where you found that first API doc page, I believe we should have it fully removed from all our public docs, so if it isn't that would be great to know.

@ynx  knows more of the technicals than me, but what he's described regarding setidentity and setidentitysource is right. One thought here though: wouldn't you have to pass in two identities for the capture? One for email address and one for the tracking ID we already know about? Otherwise, if the email is passed alone in the capture, there is no way to tie it back to the actual profile of the visitor?

**Comment by dpagini** [comment]
Yes, I believe I likely just got there by Googling "Acquia Lift Decision API".

Thanks to you both for the suggestions. I am actually pretty far along with this so far and right about where you're suggesting. I am sending in the `tc_ptid` cookie value as the `$capture_payload-> setTouchIdentifier($value);`, but that's the piece that's not working for me. It seems the email address I'm passing in is being associated with a NEW touch identifier.

Anyways, I think I can figure this stuff out, and at least work with my Lift contacts as well.

Should I leave this issue open, and maybe change it's name, to suggest a release for this library and potentially getting it included on packagist?

**Comment by kaynenh** [comment]
Hey all, thanks for jumping in here. David, I took a look at the gist you sent me and I think it's possible that you actually want to use the event 'updatePerson' vs 'updatePersonTracking'. I'm hopeful that will solve this for you.

**Comment by dpagini** [comment]
Hmm @kaynenh thanks for the suggestion, that doesn't seem to do it either. =/
Would there be any hint in the return response from the capture endpoint? It doesn't look to me like there's much there, but just throwing out ideas...

**Comment by kaynenh** [comment]
David and I have emailed outside of this and found that the issue was that he needed to set the cookie identifier as an extra identifier in his code, like so:

```
$cookie_id = '1231231231'; //set to real value
$extra_identities = [$cookie_id => 'tracking'];
$capture->setIdentities($extra_identities);
```

After that, his email address was associated to the correct person.

**Comment by ynx** [comment]
Thank you Kaynen! I have included your example in the codebase @ fa30e935ec5a17b9d67cff43ede954ccdf9d591d. Closing this issue.
